rl4co/utils/ops.py

```diff
@@ -147,7 +147,7 @@ def select_start_nodes(td, env, num_starts):
         num_starts: Number of nodes to select. This may be passed when calling the policy directly. See :class:`rl4co.models.AutoregressiveDecoder`
     """
     num_loc = env.generator.num_loc if hasattr(env.generator, "num_loc") else 0xFFFFFFFF
-    if env.name in ["tsp", "atsp"]:
+    if env.name in ["tsp", "atsp", "flp", "mcp"]:
```

Comment:
We may use the `env.select_start_nodes` as well, but this is also OK as a fallback method

Reply:
I see! I do think we may need some different ways of "selected_start_nodes" for non-routing problems since choosing all distinct starting nodes might not be reasonable for some envs.

Reply:
Yes, exactly. If the function is provided in the env, by default it will be provided

Reply:
Where can I see an example of "env.select_start_nodes being provided"?

Reply:
For example in PDP environment [here](https://github.com/ai4co/rl4co/blob/2e333356a2d20e9661b9b0793d7b5cc4b9fd53f0/rl4co/envs/routing/pdp/env.py#L222) only pickup nodes can be selected as first node.



By default, [this](https://github.com/ai4co/rl4co/blob/2e333356a2d20e9661b9b0793d7b5cc4b9fd53f0/rl4co/utils/decoding.py#L291) is called (i.e. it calls the class of RL4COEnvBase [here](https://github.com/ai4co/rl4co/blob/2e333356a2d20e9661b9b0793d7b5cc4b9fd53f0/rl4co/envs/common/base.py#L206) which is provided by the function in `utils`. If `select_start_nodes` is overwritten, then it will be taken from that environment: this is preferable and more modular, and eventually we would like to change to this behavior